analysis/pt.lean

```diff
@@ -0,0 +1,278 @@
+
+@[simp] lemma prob_eq_coerc {s : samp_space} [measurable_space s] (p : prob s) (a : set s): 
+p a =  p.to_measure a := by refl 
+
+
+lemma prob_empty {s : samp_space} [measurable_space s] (p : prob s) : p(∅) = 0 :=
+p.empty 
+
+lemma prob_univ {s : samp_space} [measurable_space s] (p : prob s) : p(univ) = 1 := 
+p.is_one
+
+lemma prob_le_1 {s : samp_space} [measurable_space s] (p : prob s) : 
+Π (a : set(s)),p(a) ≤ (1:ennreal) :=
+begin
+    intros, rewrite ← prob_univ p, apply outer_measure.mono,
+    apply subset_univ, 
+end
+
+
+lemma prob_ne_top {s : samp_space} [measurable_space s] (p : prob s):
+Π (a : set(s)), p(a) ≠ ⊤ :=
+begin
+    intros a,
+    rw ← lattice.lt_top_iff_ne_top,
+    have h : (1 : ennreal) < ⊤, {
+        rw lattice.lt_top_iff_ne_top,
+    apply ennreal.one_ne_top,
+    },
+    apply lt_of_le_of_lt (prob_le_1 p a) h,
+end
+
+lemma prob_mono {s : samp_space} [measurable_space s] (p : prob s) (a b : set s)
+(h1 : is_measurable a) (h2 : is_measurable b):
+(a ⊆ b) → (p(a) ≤ p(b)) :=  measure_theory.measure_mono 
+
+lemma prob_comp {s : samp_space} [measurable_space s] (p : prob s) 
+(a : set s)
+(h: is_measurable a): p(-a) + p(a) = 1 :=
+begin
+    intros, rewrite ← prob_univ p,
+    simp only [prob_eq_coerc],
+    rewrite ← compl_union_self a,
+    rw measure_union _ _ _,
+    apply disjoint_iff.2 (@compl_inter_self _ a),
+    apply is_measurable.compl, repeat{assumption},
+end
+
+lemma prob_union {s : samp_space} [measurable_space s] (s₁ s₂ : set s)
+(hd : disjoint s₁ s₂) (h₁ : is_measurable s₁) (h₂ : is_measurable s₂) (p : prob s) :
+  p (s₁ ∪ s₂) = p s₁ + p s₂ := measure_theory.measure_union hd h₁ h₂ 
+
+lemma prob_diff_inter {s : samp_space} [measurable_space s](p : prob s) 
+(a b : set s) (h₁ : is_measurable a) (h₂ : is_measurable b) :
+p(b ∩ -a) + p(b ∩ a) = p(b) :=
+begin
+    have h :p(b) = p(b ∩ univ),
+    by rewrite inter_univ b,
+    rewrite [h,← compl_union_self a,set.inter_distrib_left,prob_union],
+    have g : (b ∩ -a) ∩ (b ∩ a) = ∅, by rw [inter_left_comm,set.inter_assoc, compl_inter_self,inter_empty,inter_empty],
+    apply disjoint_iff.2 g,
+    {
+        rewrite ← diff_eq,
+        apply is_measurable.diff h₂ h₁,
+    },
+    apply is_measurable.inter h₂ h₁,
+end
+
+lemma prob_union_inter {s : samp_space} [measurable_space s] (p : prob s) 
+(a b : set s) (h_1 : is_measurable a) (h_2 : is_measurable b) :
+p(a ∪ b) + p(a ∩ b) = p(a) + p(b) :=
+begin 
+    have h₁ : a ∪ b = a ∪ (b ∩ -a),by
+        rewrite [set.union_distrib_left, union_compl_self a,inter_univ],
+    have h₂ : is_measurable(b ∩ -a),
+    {
+        rewrite ← diff_eq,
+        apply is_measurable.diff h_2 h_1,
+    },
+    rewrite [h₁,prob_union a _ _ h_1 h₂],
+    have h₃: a ∩ b = b ∩ a, by exact inter_comm a b,
+    rewrite [h₃,← prob_diff_inter p a b h_1 h_2],
+    simp,
+    have h₄ : a ∩ (b ∩ -a) = ∅, by tidy,
+    apply disjoint_iff.2 h₄,
+end
+
+lemma bonferroni {s : samp_space} [measurable_space s] (p : prob s)
```

Comment:
The mathlib convention is to give the lemma a name following the style guide, and to include the "classical" name in the docstring. So this would probably be called `prob_add_le_inter_add_one`.
Docstrings should go on the line before the statement
```lean
/-- The Bonferroni Lemma. -/
```